articles/node_lifecycle.md

```diff
@@ -176,7 +183,7 @@ It may be entered from any Primary State except `Finalized`, the originating sta
 
 #### Valid transitions out of ShuttingDown
 
-- If the `onShutdown` callback succeeds the node will transition to `Finalized`.
+- If the `onShutdown` callback succeeds or results in a failure code, the node will transition to `Finalized`.
```

Comment:
Is this correct? with expected failure seems that it cannot shutdown in some reason, so node should  transition back to the previous primary state?

Reply:
That's how it is in the current implementation.
https://github.com/ros2/rcl/blob/b786cea3489c829987238a29afdd80ae70bf27ed/rcl_lifecycle/src/default_state_machine.c#L570
```
  // register transition from shutting down to finalized
  {
    rcl_lifecycle_transition_t rcl_transition_on_shutdown_failure = {
      rcl_lifecycle_transition_failure_label,
      lifecycle_msgs__msg__Transition__TRANSITION_ON_SHUTDOWN_FAILURE,
      shuttingdown_state, finalized_state
    };
    ret = rcl_lifecycle_register_transition(
      transition_map,
      rcl_transition_on_shutdown_failure,
      allocator);
    if (ret != RCL_RET_OK) {
      return ret;
    }
  }
```

Reply:
yes, actually based on the current implementation, this description is correct. but it is weird to me, shutdown is failure but transition succeeds, i guess that it needs to get back in the previous state to retry?

Reply:
If it's shutting down, then a failure is probably "something went wrong but the node is dead now anyone" rather than "I couldn't shut down so I'm going to live forever".

Reply:
@gbiggs 

ah, that makes more sense and works for actual use cases. thanks for explanation 👍